crates/bevy_render/src/view/mod.rs

```diff
@@ -116,78 +117,107 @@ pub struct ViewUniformOffset {
-            ViewMainTexture::Sdr {
-                texture,
-                sampled_texture,
-            } => (texture, sampled_texture),
-        };
-        match sampled {
-            Some(sampled_target) => RenderPassColorAttachment {
-                view: sampled_target,
-                resolve_target: Some(target),
-                ops,
-            },
-            None => RenderPassColorAttachment {
-                view: target,
-                resolve_target: None,
+        match &self.main_textures.sampled {
+            Some(sampled_texture) => RenderPassColorAttachment {
+                view: sampled_texture,
+                resolve_target: Some(self.main_texture()),
                 ops,
             },
+            None => self.get_unsampled_color_attachment(ops),
         }
     }
 
+    /// Retrieve an "unsampled" color attachment using [`Self::main_texture`].
     pub fn get_unsampled_color_attachment(
         &self,
         ops: Operations<Color>,
     ) -> RenderPassColorAttachment {
         RenderPassColorAttachment {
-            view: match &self.main_texture {
-                ViewMainTexture::Hdr { hdr_texture, .. } => hdr_texture,
-                ViewMainTexture::Sdr { texture, .. } => texture,
-            },
+            view: self.main_texture(),
             resolve_target: None,
             ops,
         }
     }
+
+    /// The "main" unsampled texture.
+    pub fn main_texture(&self) -> &TextureView {
+        if self.main_texture.load(Ordering::SeqCst) == 0 {
+            &self.main_textures.a
+        } else {
+            &self.main_textures.b
+        }
+    }
+
+    /// The "main" sampled texture.
+    pub fn sampled_main_texture(&self) -> Option<&TextureView> {
+        self.main_textures.sampled.as_ref()
+    }
+
+    #[inline]
+    pub fn main_texture_format(&self) -> TextureFormat {
+        self.main_texture_format
+    }
+
+    /// Returns `true` if and only if the main texture is [`Self::TEXTURE_FORMAT_HDR`]
+    #[inline]
+    pub fn is_hdr(&self) -> bool {
+        self.main_texture_format == ViewTarget::TEXTURE_FORMAT_HDR
+    }
+
+    /// The final texture this view will render to.
+    #[inline]
+    pub fn out_texture(&self) -> &TextureView {
+        &self.out_texture
+    }
+
+    /// The format of the final texture this view will render to
+    #[inline]
+    pub fn out_texture_format(&self) -> TextureFormat {
+        self.out_texture_format
+    }
+
+    /// This will start a new "post process write", which assumes that the caller
+    /// will write the [`PostProcessWrite`]'s `source` to the `destination`.
+    /// `source` is the "current" main texture. This will internally flip this
+    /// [`ViewTarget`]'s main texture to the `destination` texture, so the caller
+    /// _must_ ensure `source` is copied to `destination`, with or without modifications.
+    /// Failing to do so will cause the current main texture information to be lost.
```

Comment:
Very minor nit, but having a [brief summary line](https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html?highlight=summary#documenting-components) as a first paragraph makes this render more nicely in rustdoc.